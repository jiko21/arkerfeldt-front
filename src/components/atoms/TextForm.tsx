/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { FC, InputHTMLAttributes } from 'react';
import { Color } from '@/const/color';

const inputStyle = css`
  border: 1px solid ${Color.BLACK};
  border-radius: 6px;
  height: 30px;
  min-width: 240px;
  width: 100%;
  ::placeholder {
    color: ${Color.LIGHT};
  }
`;

interface IProps extends InputHTMLAttributes<HTMLInputElement> {
  testId?: string;
}

const TextForm: FC<IProps> = ({ testId, ...props }) => (
  <input css={inputStyle} {...props} data-testid={testId} />
);

export default TextForm;
