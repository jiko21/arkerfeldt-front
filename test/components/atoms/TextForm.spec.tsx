import TextForm from '@/components/atoms/TextForm';
import { fireEvent, render, screen } from '@testing-library/react';

describe('TextForm.tsx', () => {
  let onChangeMock: jest.Mock;
  beforeEach(() => {
    onChangeMock = jest.fn();
  });

  afterEach(() => {
    onChangeMock.mockClear();
  });
  it('should correctly render', () => {
    const { container } = render(
      <TextForm type={'email'} value={'sample'} onChange={onChangeMock} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should correctly render when text is not empty', () => {
    const { container } = render(
      <TextForm type={'email'} value={'sample'} onChange={onChangeMock} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should correctly onChange when value changed', async () => {
    const TEST_ID = 'TEST_ID';
    render(
      <TextForm
        type={'email'}
        value={''}
        onChange={onChangeMock}
        testId={TEST_ID}
      />,
    );
    const input = await screen.findByTestId(TEST_ID);
    fireEvent.change(input, {
      target: {
        value: 'test',
      },
    });
    expect(onChangeMock).toBeCalled();
  });
});
