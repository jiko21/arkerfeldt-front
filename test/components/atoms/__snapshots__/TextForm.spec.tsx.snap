// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextForm.tsx should correctly render 1`] = `
.emotion-0 {
  border: 1px solid #2d2d2d;
  border-radius: 6px;
  height: 30px;
  min-width: 240px;
  width: 100%;
}

.emotion-0::-webkit-input-placeholder {
  color: #878686;
}

.emotion-0::-moz-placeholder {
  color: #878686;
}

.emotion-0:-ms-input-placeholder {
  color: #878686;
}

.emotion-0::placeholder {
  color: #878686;
}

<div>
  <input
    class="emotion-0"
    type="email"
    value="sample"
  />
</div>
`;

exports[`TextForm.tsx should correctly render when text is not empty 1`] = `
.emotion-0 {
  border: 1px solid #2d2d2d;
  border-radius: 6px;
  height: 30px;
  min-width: 240px;
  width: 100%;
}

.emotion-0::-webkit-input-placeholder {
  color: #878686;
}

.emotion-0::-moz-placeholder {
  color: #878686;
}

.emotion-0:-ms-input-placeholder {
  color: #878686;
}

.emotion-0::placeholder {
  color: #878686;
}

<div>
  <input
    class="emotion-0"
    type="email"
    value="sample"
  />
</div>
`;
